ethcore/types/src/transaction/transaction.rs

```diff
@@ -462,6 +462,24 @@ impl SignedTransaction {
 		self.sender
 	}
 
+	/// Returns transaction receiver, if any
+	pub fn receiver(&self) -> Option<Address> {
+		match self.transaction.unsigned.action {
+			Action::Create => None,
+			Action::Call(receiver) => Some(receiver),
+		}
+	}
+
+	// Reference to the plain text transaction
+	pub fn as_unsigned(&self) -> &Transaction {
+		self.transaction.as_unsigned()
+	}
+
+	/// Returns transaction hash
+	pub fn hash(&self) -> H256 {
```

Comment:
Not needed, we have `fn hash` in `UnverifiedTransaction` and there is a `Deref` to it.